sample/src/main/kotlin/com/slack/circuit/sample/home/HomePresenter.kt

```diff
@@ -116,33 +145,89 @@ class HomeUiFactory @Inject constructor() : Ui.Factory {
 private fun homeUi() =
   ui<HomeScreen.State, HomeScreen.Event> { state, events -> HomeContent(state, events) }
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun HomeContent(state: HomeScreen.State, eventSink: (HomeScreen.Event) -> Unit) {
-  Scaffold(
-    modifier = Modifier.navigationBarsPadding().systemBarsPadding().fillMaxWidth(),
-    topBar = {
-      CenterAlignedTopAppBar(
-        title = {
-          Text("Adoptables", fontSize = 22.sp, color = MaterialTheme.colorScheme.onPrimaryContainer)
-        },
-        colors =
-          TopAppBarDefaults.centerAlignedTopAppBarColors(
-            containerColor = MaterialTheme.colorScheme.primaryContainer
+  val modalState =
+    rememberModalBottomSheetState(
+      initialValue =
+        if (state.petListFilterState.showBottomSheet) ModalBottomSheetValue.Expanded
+        else ModalBottomSheetValue.Hidden
+    )
+
+  // Monitor bottom sheet state and emit event whenever the user dismisses the modal
+  LaunchedEffect(modalState) {
```

Comment:
Comment for posterity - I think this case would be a good use case for a future result API instead of treating the modal as composite state